src/misc/cross.cpp

```diff
@@ -782,6 +783,91 @@ std::string cfstr_to_string(CFStringRef source)
 	return lang;
 }
 
+#ifndef WIN32
+uint16_t xattr_string_to_fattr(const char* xattr)
+{
+	uint16_t fattr = 0;
+
+	if (strchr(xattr, 'R')) {
+		fattr |= DOS_ATTR_READ_ONLY;
+	}
+	if (strchr(xattr, 'H')) {
+		fattr |= DOS_ATTR_HIDDEN;
+	}
+	if (strchr(xattr, 'S')) {
+		fattr |= DOS_ATTR_SYSTEM;
+	}
+	if (strchr(xattr, 'A')) {
+		fattr |= DOS_ATTR_ARCHIVE;
+	}
+
+	return fattr;
+}
+
+void fattr_to_xattr_string(uint16_t fattr, char* outbuf, size_t outsize)
```

Comment:
Requiring the user of the function to pass in `outsize` is very oldschool; we don't do that. Please use `std::string`. This is not performance intensive code, and even if it was, we can still fully take advantage of the safe and easy to use `std::string` interface and use string views.

Reply:
What's the point of that if it's immediately converted to C string anyway for writing xattr?

Reply:
Consistency. Safety. But moot point if you're going the hex value route.